src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -63,13 +62,25 @@ internal class HttpInListener : ListenerHandler
 #endif
         private readonly PropertyFetcher<Exception> stopExceptionFetcher = new("Exception");
         private readonly AspNetCoreInstrumentationOptions options;
+        private readonly bool emitOldAttributes;
+        private readonly bool emitNewAttributes;
 
         public HttpInListener(AspNetCoreInstrumentationOptions options)
             : base(DiagnosticSourceName)
         {
             Guard.ThrowIfNull(options);
 
             this.options = options;
+
+            if (this.options.HttpSemanticConvention.HasFlag(HttpSemanticConventionHelper.HttpSemanticConvention.Old))
+            {
+                this.emitOldAttributes = true;
+            }
```

Comment:
This could be simplified:

```suggestion
this.emitOldAttributes = this.options.HttpSemanticConvention.HasFlag(HttpSemanticConventionHelper.HttpSemanticConvention.Old);
```